sys/parsoid_bucket.js

```diff
@@ -0,0 +1,241 @@
+"use strict";
+
+/**
+ * Key-rev-value bucket handler
+ */
+
+const P = require('bluebird');
+const uuid = require('cassandra-uuid').TimeUuid;
+const mwUtil = require('../lib/mwUtil');
+const HyperSwitch = require('hyperswitch');
+const HTTPError = HyperSwitch.HTTPError;
+const URI = HyperSwitch.URI;
+const spec = HyperSwitch.utils.loadSpec(`${__dirname}/parsoid_bucket.yaml`);
+
+// Format a revision response. Shared between different ways to retrieve a
+// revision (latest & with explicit revision).
+function returnRevision(req) {
+    return (dbResult) => {
+        if (dbResult.body && dbResult.body.items && dbResult.body.items.length) {
+            const row = dbResult.body.items[0];
+            const headers = {
+                etag: mwUtil.makeETag(row.rev, row.tid),
+                'content-type': row['content-type']
+            };
+            return {
+                status: 200,
+                headers,
+                body: row.value
+            };
+        } else {
+            throw new HTTPError({
+                status: 404,
+                body: {
+                    type: 'not_found',
+                    uri: req.uri,
+                    method: req.method
+                }
+            });
+        }
+    };
+}
+
+class ParsoidBucket {
+    createBucket(hyper, req) {
+        const rp = req.params;
+        return P.all([
+            hyper.put({
+                uri: new URI([rp.domain, 'sys', 'table-ng', 'html-ng']),
```

Comment:
Oh, and `table-ng` should probably be made a constant and used as such, since once we switch all of the use cases to the new storage design `/sys/table/` will no longer be used and we will be able to safely rename this one.